### PR TITLE
Dbt on composer

### PR DESCRIPTION
**Pull request by adiwijaya**
Fixing the TODO list in dbt-on-composer example code. 
[/issues/712](https://github.com/GoogleCloudPlatform/professional-services/issues/712)

Fixing the service account key on GCS bucket usage. Modified to use short-live service account key, managed in Cloud Build steps.

### CODE REVIEW

File: examples/dbt-on-cloud-composer/basic/dbt-project/cloudbuild.yaml [hunk 1]
```diff
@@ -37,27 +38,47 @@ steps:
...
+- name: gcr.io/cloud-builders/gcloud
   id: Get SA key
+  entrypoint: "bash"
   args:
-  - cp
-  - gs://pso-dbt-airflow-demo-bucket/key/pso-dbt-airflow-demo-a87a347b6390.json
-  - key/pso-dbt-airflow-demo-a87a347b6390.json
+  - "-c"
+  - |
+    gcloud iam service-accounts keys create sa-key.json --iam-account=${_DBT_SERVICE_ACCOUNT}
```
Comment (iht) [inline examples/dbt-on-cloud-composer/basic/dbt-project/cloudbuild.yaml:50]: Is it essential to create a key for the service account to use this example? We recommend not exporting keys, and using impersonation instead.

**Review by iht — CHANGES_REQUESTED** [review]
Please see comment about exporting keys for service accounts.

File: examples/dbt-on-cloud-composer/basic/dbt-project/cloudbuild.yaml (hunk 1 above)
Comment (adiwijaya) [reply examples/dbt-on-cloud-composer/basic/dbt-project/cloudbuild.yaml:50]: hi @iht , in this example the service account key is needed to run dbt CLI.
The [CLI is accepting SA key file](https://docs.getdbt.com/reference/warehouse-profiles/bigquery-profile#service-account-file)

Comment (adiwijaya) [reply examples/dbt-on-cloud-composer/basic/dbt-project/cloudbuild.yaml]: Hi, I pushed another commit. I just knew dbt has option to directly use "impersonate_service_account" parameter in the run.

**Comment by tims** [comment]
Looks good, @iht reassigning to you since you've already started your review.

**Review by iht — APPROVED** [review]
LGTM, thanks for your contribution to the repo!

**Pull request merged**